### PR TITLE
api: run DeleteCmd when KubernetesApply object deleted

### PR DESCRIPTION
**Pull request by milas**
When a custom `DeployCmd` is provided, a corresponding `DeleteCmd`
is needed to handle cleanup.

Although Tilt is able to track + diff the managed entities to determine
what to delete (which is still the case when using YAML), the
`DeployCmd` might create additional state, e.g. a Helm release in
Tiller/`Secret`, so this provides the tool with the means to clean
up the external state _as well as all relevant entities in the
cluster_.

### CODE REVIEW

File: internal/controllers/core/kubernetesapply/reconciler.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ import (
 	"github.com/tilt-dev/tilt/pkg/model"
 )
 
+type garbageCollectMeta struct {
```
Comment (nicks) [inline internal/controllers/core/kubernetesapply/reconciler.go:39]: what does 'meta' signify here?

Comment (nicks) [reply internal/controllers/core/kubernetesapply/reconciler.go:39]: maybe it should be called DeleteSpec or smth

File: internal/controllers/core/kubernetesapply/reconciler.go [hunk 2]
```diff
@@ -524,46 +523,72 @@ func (r *Reconciler) updateResult(nn types.NamespacedName, result *Result) []k8s
...
 	// Go through all the results, and check to see which objects
 	// we're not managing anymore.
-	var toDeleteMap objectRefSet
-	if existing != nil {
-		toDeleteMap = existing.AppliedObjects.clone()
-		for _, result := range r.results {
-			for objRef := range result.AppliedObjects {
-				delete(toDeleteMap, objRef)
-			}
+	// TODO(milas): in the case that the KA object was deleted, should we respect `tilt.dev/down-policy`?
```
Comment (nicks) [inline internal/controllers/core/kubernetesapply/reconciler.go:542]: i'd vote yes, though agreed that the current code doesn't respect this

**Review by nicks — APPROVED** [review]

File: internal/controllers/core/kubernetesapply/reconciler.go (hunk 1 above)
Comment (milas) [reply internal/controllers/core/kubernetesapply/reconciler.go]: 'meta' signifies I was having a hard time naming this object 😉 

(Renamed to `deleteSpec`, thanks for suggestion)

**Pull request merged**